### PR TITLE
[21053] Move DR TypeConsistencyEnforcement & DataRepresentation from TypeConsistency to DataReaderQos

### PR DESCRIPTION
**Pull request by JesusPoderoso**
## Description

This PR updates the usage of DR qos DataRepresentation according to the changes introduced in:

Related implementation PR:
* eProsima/Fast-DDS#4823

## Contributor Checklist

- [x] Commit messages follow the project guidelines. 
- [x] Changes do not break current interoperability.
- __N/A__ New feature has been documented/Current behavior is correctly described in the documentation. 
    
- __N/A__ Applicable backports have been included in the description.

## Reviewer Checklist

- [x] The PR has a milestone assigned.
- [x] The title and description correctly express the PR's purpose.
- [x] Check contributor checklist is correct.
- [x] CI passes without warnings or errors.

### CODE REVIEW

**Comment by JesusPoderoso** [comment]
CI passed locally, it will not pass until Fast DDS PR is merged

**Review by Mario-DL — APPROVED** [review]
LGTM with green CI. We will wait for `Fast DDS` PR to be merged.

**Comment by Mario-DL** [comment]
@richiprosima please test this

**Review by Mario-DL — APPROVED** [review]
LGTM

**Pull request merged**
